lib/src/rules/unnecessary_constructor_name.dart

```diff
@@ -70,6 +70,7 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitConstructorDeclaration(ConstructorDeclaration node) {
+    if (node.parent is ExtensionTypeDeclaration) return;
```

Comment:
I don't understand why we wouldn't check constructors in extension types.

If anything, I think this lint needs to check the primary constructor (which it currently probably doesn't do).

Reply:
You can't redefine the unnamed constructor in an extension type so my thinking is this lint should never fire.

Reply:
(OK. I can see I need to update this in case the primary constructor is named. Updating.)

Reply:
But you can use `.new` in a constructor declaration (whether it's the primary constructor or not), and I think this lint ought to flag that given that it's flagging the same use in constructors in classes. Examples:

```
extension type E.new(int i) {}
```

and

```
extension type E.name(int i) {
  E.new(this.i);
}
```

Reply:
Thanks for your help.  I think I've got it right now!